### ISSUE TITLE
ipdiscover compilation should use $Config::Config{ccflags} and other MakeMaker flags

### ISSUE THREAD

**Issue by guimard** [OCSInventory-NG/UnixAgent#140]
In Makefile.PL, ipdiscover is compiled using:
```
my $cc=$Config::Config{cc};
    system("$cc resources/ipdiscover/ipdiscover.c -o ipdiscover");
```
It **should** use $Config::Config{ccflags}

### General informations
Operating system : Any
Perl version : Any

### OCS Inventory informations
Unix agent version : 2.4.1

**Comment by guimard** [comment]
More things to do: a classic package compilation provides some args to Makefile.PL, Debian example:
```
perl -I. Makefile.PL INSTALLDIRS=vendor "OPTIMIZE=-g -O2 -fdebug-prefix-map=/home/xavier/dev/debian/ocsinventory-agent=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2" "LD=x86_64-linux-gnu-gcc -g -O2 -fdebug-prefix-map=/home/xavier/dev/debian/ocsinventory-agent=. -fstack-protector-strong -Wformat -Werror=format-security -Wl,-z,relro -Wl,-z,now"
```
This build a Makefile that uses this args.

`ipdiscover` is built directly by calling `$cc=$Config::Config{cc}` without any vendor arguments neither ExtUtils::MakeMaker parameters. This produce a basic ELF exe without any security flags. This is considered for Debian as a security issue. For now, I'm using this patch (Debian specific):
```
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -65,7 +65,8 @@
 #Ugly hack to include ipdiscover binary compilation and install fo Linux systems
 if ($^O =~ /^linux$/i && can_cc()) {
        my $cc=$Config::Config{cc};
-       system("$cc resources/ipdiscover/ipdiscover.c -o ipdiscover");
+       my $fl=$Config::Config{ccflags};
+       system("$cc $fl `dpkg-buildflags --get CPPFLAGS` `dpkg-buildflags --get CFLAGS` `dpkg-buildflags --get LDFLAGS` resources/ipdiscover/ipdiscover.c -o ipdiscover");
        if (-f 'ipdiscover') {
          install_script 'ipdiscover';
        }
```

**Comment by remicollet** [comment]
I don't think calling `dpkg-buildflags` is acceptable

**Comment by guimard** [comment]
This is an example, not a fix for you

**Comment by guimard** [comment]
You just have to use at least $(LD) and $(OPTIMIZE)

**Comment by guimard** [comment]
Or build ipdiscover using Makefile, not Makefile.PL

**Comment by fbomj** [comment]
Fix with PR #345
